AudioDeviceExample/ViewController.swift

```diff
@@ -0,0 +1,466 @@
+            self.setNeedsUpdateOfHomeIndicatorAutoHidden()
+        }
+        self.setNeedsStatusBarAppearanceUpdate()
+
+        self.navigationController?.setNavigationBarHidden(inRoom, animated: true)
+    }
+
+    func dismissKeyboard() {
+        if (self.roomTextField.isFirstResponder) {
+            self.roomTextField.resignFirstResponder()
+        }
+    }
+
+    func logMessage(messageText: String) {
+        messageLabel.text = messageText
+
+        if (messageLabel.alpha < 1.0) {
+            self.messageLabel.isHidden = false
+            UIView.animate(withDuration: 0.4, animations: {
+                self.messageLabel.alpha = 1.0
+            })
+        }
+
+        // Hide the message with a delay.
+        self.messageTimer?.invalidate()
+        if #available(iOS 10.0, *) {
+            let timer = Timer.init(timeInterval: TimeInterval(6), repeats: false) { (timer) in
+                if (self.messageLabel.isHidden == false) {
+                    UIView.animate(withDuration: 0.6, animations: {
+                        self.messageLabel.alpha = 0
+                    }, completion: { (complete) in
+                        if (complete) {
+                            self.messageLabel.isHidden = true
+                        }
+                    })
+                }
+            }
+
+            self.messageTimer = timer
+            RunLoop.main.add(timer, forMode: .commonModes)
+        } else {
+            // TODO: iOS 9.x support.
+        }
+    }
+
+    func prepareLocalMedia() {
+        /*
+         * The important thing to remember when using a custom TVIAudioDevice is that the device must be set
+         * before performing any other actions with the SDK (such as creating Tracks, or connecting to a Room).
+         */
+        TwilioVideo.audioDevice = ExampleCoreAudioDevice.init();
+
+        /*
+         * ExampleCoreAudioDevice is a playback only device. Because of this, any attempts to create a
+         * TVILocalAudioTrack will result in an exception being thrown, and nil being returned. In this example we will
+         * only share video (where available) and not audio.
+         */
+        if (TVICameraCapturer.isSourceAvailable(TVICameraCaptureSource.frontCamera)) {
+
+            // We will render the camera using TVICameraPreviewView.
+            camera = TVICameraCapturer(source: .frontCamera, delegate: self, enablePreview: true)
+            localVideoTrack = TVILocalVideoTrack.init(capturer: camera!)
+
+            if (localVideoTrack == nil) {
+                logMessage(messageText: "Failed to create video track!")
+            } else {
+                logMessage(messageText: "Video track created.")
+
+                if let preview = camera?.previewView {
+//                    let tap = UITapGestureRecognizer(target: self, action: #selector(ViewController.recognizeLocalAudio))
+//                    preview.addGestureRecognizer(tap)
+                    view.addSubview(preview);
```

Comment:
I wasn't sure if tapping should do anything. There is no audio track to mute, and so far the camera is locked on front only.

Reply:
For now I've removed tap gesture handling.